src/main/java/spoon/support/reflect/declaration/CtRecordComponentImpl.java

```diff
@@ -36,18 +39,45 @@ public class CtRecordComponentImpl extends CtNamedElementImpl implements CtRecor
 	public CtMethod<?> toMethod() {
 		CtMethod<?> method = this.getFactory().createMethod();
 		method.setSimpleName(getSimpleName());
-		method.setType((CtTypeReference) getType());
+		method.setType(getType());
```

Comment:
```suggestion
		method.setType(getType().clone());
```
Otherwise the method will have an invalid parent, see #5870